evalml/data_checks/invalid_targets_data_check.py

```diff
@@ -67,10 +76,14 @@ def validate(self, X, y):
         unique_values = value_counts.index.tolist()
 
         if self.problem_type == ProblemTypes.BINARY and len(value_counts) != 2:
+            if self.n_unique is None:
+                details = {"target_values": unique_values}
+            else:
+                details = {"target_values": unique_values[:min(self.n_unique, len(unique_values))]}
```

Comment:
nit-pick: I think the `min` is redundant because if the end of the slice is larger than the list length, you'll get the whole list anyways.

Reply:
Ah, I think if self.n_unique > len(unique_values), you get an `IndexError`? Though I think I should have written len(unique_values)-1` on second thought.

Reply:
I don't think you'd get the index error when you slice a list. But it's a minor point and what you have now works and is clear!

Reply:
Woah, TIL slicing is magical and will just give you the whole list, thanks @freddyaboulton :')

(Will likely still keep this/something similar for clarity sake though)